source/h1_encoder.c

```diff
@@ -55,6 +58,13 @@ static int s_scan_outgoing_headers(const struct aws_http_message *message, size_
             return AWS_OP_ERR;
         }
     }
+    if (expects_no_body) {
+        /* no body should follow, no matter what the headers are */
```

Comment:
I think we need a more specific word than "expect".
Sometimes we don't expect a body, so we ignore body headers.
Other times we don't expect a body, so body headers are forbidden.
Maybe more specific terms like: "body_headers_forbidden" and "body_headers_ignored"